### ISSUE TITLE
Process keeps running?

### ISSUE THREAD

**Issue by coderplus** [eclecticlogic/whisper#5]
I'm trying to run the below simple Java code 

    package chapters.configuration;
    import org.slf4j.Logger;
    import org.slf4j.LoggerFactory;

    public class MyApp1 {
      final static Logger logger = LoggerFactory.getLogger(MyApp1.class);

       public static void main(String[] args) {
	  
	      for(int i=0;i<10;i++){
	    	  logger.error("NPE", new NullPointerException());
	  }
	  for(int i=0;i<10;i++){
        logger.error("Exiting application."+i);
	  }
      }
    }

using the logback configuration file

    <?xml version="1.0" encoding="UTF-8" ?>    
    <configuration scan="true" scanPeriod="1 minute" debug="true">
    	<contextName>testContext</contextName>
    
    	
    	<appender name="CONSOLE" class="ch.qos.logback.core.ConsoleAppender">
    		<layout class="ch.qos.logback.classic.PatternLayout">
    			<pattern>%d{yyyy-MM-dd HH:mm:ss.SSS} [%X{ApplicationId}] [%X{UnitOfWorkId}] [%thread] %-5level %logger{36} - %msg%n</pattern>
    		</layout>
    	</appender>
    
    	<appender name="errorEmail" class="ch.qos.logback.classic.net.SMTPAppender">
    		<SMTPHost>localhost</SMTPHost>
    		<SMTPPort>25</SMTPPort>
    		
    		<to>test@test.com</to>
    		<from>test@localhost</from>
    		
    		<subject>%logger{20} - %m</subject>
    		
    		<layout class="ch.qos.logback.classic.PatternLayout">
    			<pattern>%date %-5level %logger{35} - %message%n</pattern>
    		</layout>
    	</appender>
    	
    		
    	<appender name="whisper"
    		class="com.eclecticlogic.whisper.logback.WhisperAppender">
    		
    		<filter class="ch.qos.logback.classic.filter.LevelFilter">
    			<level>ERROR</level>
    			<onMatch>ACCEPT</onMatch>
    			<onMismatch>DENY</onMismatch>
    		</filter>
    		
    		<digestLoggerName>digest.appender.logger</digestLoggerName>
    		
    		<suppressAfter>3 in 5 minutes</suppressAfter>
    		 
    		<expireAfter>4 minutes</expireAfter>
    		
    		<digestFrequency>10 seconds</digestFrequency>
    		
    		
    		<appender-ref ref="errorEmail" />
    	</appender>
    	
    	
    	<appender name="digest" class="ch.qos.logback.core.ConsoleAppender">
    		
    		<encoder>
    			<pattern>%d{HH:mm:ss.SSS} [%thread] %-5level %logger{36} - %X{whisper.digest.subject} - %msg%n</pattern>
    		</encoder>
    	</appender>
    	
    	
    	<logger name="digest.appender.logger" level="error" additivity="false">
    		<appender-ref ref="digest" />
    	</logger>
    
    	<root level="info">
    		<appender-ref ref="CONSOLE" />
    		<appender-ref ref="whisper" />
    	</root>
    </configuration>
    

I was expecting the app to exit after logging "Exiting application.9", but it hangs around like for ever.

**Comment by eclecticlogic** [comment]
I was able to reproduce the behavior but it seems like even if you remove all references to Whisper the application still hangs. Moreover, if you replace the SMTP appender with a console appender, the application correctly shuts down. Digging a little more, I found that in the SMTPAppenderBase class of logback, there is a call to context.getExecutorService() (in the append method). ContextBase which implements Context sets up an executor that doesn't use daemon threads but has a call to shutdown/stop from ContextBase's stop. I can see calls to stop from RollingFileAppender and SiftingAppender but not from SMTPAppender.

If you agree with my analysis above, may I request you to kindly raise the issue with logback?
